Mathlib/Data/Set/ToFunction.lean

```diff
@@ -0,0 +1,64 @@
+/-
+Copyright (c) 2024 Martin Dvorak. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Martin Dvorak
+-/
+import Mathlib.Data.Set.Defs
+import Mathlib.Logic.ExistsUnique
+
+/-!
+# Set to function
+
+This file provides API for sets that define functions.
+
+## Main definitions
+- `Set.asPartialFun` converts `Set (α × α)` to `α → Option α`.
+- `Set.asFun` converts `Set (α × α)` to `α → α` if possible.
+
+-/
+
+variable {α : Type*}
+
+namespace Set
+
+section set_as_partial_function
+
+/-- Does given set on `α × α` represent a partial function? -/
```

Comment:
This isn't a very good docstring, as it essentially just clarifies the name `isPartialFun` without explaining what it *means*. I'd write something like
```suggestion
/-- A set `s : Set (α × α)` represents a partial function when for every `x : α` there's at most
one `y : α` with `(x, y) ∈ s`. -/
```